clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -315,12 +318,28 @@ protected RegisterTaskDefinitionRequest makeTaskDefinitionRequest(
 
     if (!StringUtils.isEmpty(description.getLaunchType())) {
       request.setRequiresCompatibilities(Arrays.asList(description.getLaunchType()));
+
+      if (FARGATE.equals(description.getLaunchType())) {
+        request.setExecutionRoleArn(ecsServiceRole);
+        request.setCpu(description.getComputeUnits().toString());
+        request.setMemory(description.getReservedMemory().toString());
+      }
     }
 
-    if (FARGATE_LAUNCH_TYPE.equals(description.getLaunchType())) {
-      request.setExecutionRoleArn(ecsServiceRole);
-      request.setCpu(description.getComputeUnits().toString());
-      request.setMemory(description.getReservedMemory().toString());
+    if (description.getCapacityProviderStrategies() != null
+        && !description.getCapacityProviderStrategies().isEmpty()) {
+      request.setRequiresCompatibilities(Arrays.asList(EC2, FARGATE));
```

Comment:
we shouldn't do this here, b/c if the capacity providers don't include any `FARGATE` ones then task def registration will fail b/c you're saying `FARGATE` is required.

suggest we instead set this within the loop on 333, and only to `FARGATE` if the CP is a FG type. This field will [default to EC2](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RegisterTaskDefinition.html#ECS-RegisterTaskDefinition-request-requiresCompatibilities) in the event it's empty.

Reply:
Updated the logic such that it adds both the EC2 requires compatibilities and the Fargate requires compatibilities if Fargate is specified

Reply:
You can only have one type of capacity per ECS service (Either EC2/EC2_SPOT or FARGATE/FARGATE_SPOT, not both). I've updated the PR to only add one or the other.

Reply:
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cluster-capacity-providers.html - the last sentence under capacity providers